src/main/scala/com/microsoft/azure/iot/iothubreact/sinks/OffsetCommitSink.scala

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+package com.microsoft.azure.iot.iothubreact.sinks
+
+import java.util.concurrent.CompletionStage
+import java.util.concurrent.atomic.AtomicLong
+
+import akka.Done
+import akka.japi.function.Procedure
+import akka.stream.javadsl.{Sink ⇒ JavaSink}
+import akka.stream.scaladsl.{Sink ⇒ ScalaSink}
+import com.microsoft.azure.iot.iothubreact.checkpointing.OffsetLoader
+import com.microsoft.azure.iot.iothubreact.checkpointing.backends.CheckpointBackend
+import com.microsoft.azure.iot.iothubreact.config.{Configuration, IConfiguration}
+import com.microsoft.azure.iot.iothubreact.{Logger, MessageFromDevice}
+
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.Future
+
+final case class OffsetCommitSink(parallelism: Int, backend: CheckpointBackend, config: IConfiguration) extends ISink[MessageFromDevice] with Logger {
```

Comment:
what would you think about renaming this class to `OffsetSaveSink` ? It would sound more in line with the `SaveOffsetOnPull` option, i.e. we use the "save" verb in both cases.